src/Parser/Priest/Shared/Modules/Features/DesperatePrayer.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import ExpandableStatisticBox from 'Main/ExpandableStatisticBox';
+
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+
+import Analyzer from 'Parser/Core/Analyzer';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import Combatants from 'Parser/Core/Modules/Combatants';
+
+class DesperatePrayer extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+  };
+
+  desperatePrayerUsages = [];
+  currentDesperatePrayerUsage;
+  deathsWithDPReady = 0;
+
+  on_byPlayer_applybuff(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.currentDesperatePrayerUsage = {
+      DamageTaken: 0,
+      HPWhenUsed: 0,
+      MaxHPWhenUsed: 0,
+    };
+  }
+
+  on_toPlayer_heal(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.currentDesperatePrayerUsage.HPWhenUsed = event.hitPoints - event.amount;
+    this.currentDesperatePrayerUsage.MaxHPWhenUsed = event.maxHitPoints;
+  }
+
+  on_toPlayer_damage(event) {
+    if(!this.combatants.selected.hasBuff(SPELLS.DESPERATE_PRAYER.id)) {
+      return;
+    }
+    this.currentDesperatePrayerUsage.DamageTaken += event.amount;
+  }
+
+  on_byPlayer_removebuff(event) {
+    if (event.ability.guid !== SPELLS.DESPERATE_PRAYER.id) {
+      return;
+    }
+    this.desperatePrayerUsages.push(this.currentDesperatePrayerUsage);
+  }
+
+  on_toPlayer_death(event) {
+    if(!this.spellUsable.isOnCooldown(SPELLS.DESPERATE_PRAYER.id)){
+      this.deathsWithDPReady++;
+    }
+  }
+
+  statistic() {
+    return (
+      <ExpandableStatisticBox
+        value={`${this.desperatePrayerUsages.length}`}
+        label={`Desperate Prayer Usage(s)`}
+        icon={<SpellIcon id={SPELLS.DESPERATE_PRAYER.id} />}>
+        <table className="table table-condensed">
+          <thead>
+            <tr>
+              <th>Cast</th>
+              <th>Damage Taken</th>
+              <th>Health When Used</th>
+            </tr>
+          </thead>
+          <tbody>
+            {
+              this.desperatePrayerUsages
+                .map((dp, index) => (
+                  <tr key={index}>
+                    <th scope="row">{ index + 1 }</th>
+                    <td>{ formatPercentage(dp.DamageTaken / dp.MaxHPWhenUsed) } %</td>
+                    <td>{ formatPercentage(dp.HPWhenUsed / dp.MaxHPWhenUsed) } %</td>
+                  </tr>
+                ))
+            }
+          </tbody>
+        </table>
+      </ExpandableStatisticBox>
+    );
+  }
+
+  suggestions(when) {
+    when(this.deathsWithDPReady).isGreaterThan(0)
+      .addSuggestion((suggest, actual, recommended) => {
+        return suggest(<span>You died with <SpellLink id={SPELLS.DESPERATE_PRAYER.id} /> on cooldown.</span>)
```

Comment:
Please use the `React.Fragment` component instead of `span`s when you don't need the wrapper for any other reason than to make React happy.

Reply:
Shouldn't this be the opposite; "You died with DP available"?

Reply:
The `DeathTracker` module already complains about dying, how about extending that with a suggestion to use DP when available instead of making an all new suggestion?

Reply:
I can see the redundancy here, but I feel like having the "You were dead for x % of the fight" statistic and the deaths with DP available portrays 2 different things. 

I'm open to suggestion on how both informations can be displayed in a clear way

Reply:
DeathTracker could reveal this information, it doesn't have to be plain text.